hs_core/views/utils.py

```diff
@@ -1242,7 +1207,8 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
     unzip_temp_folder = ''
     try:
         # unzip to a temporary folder first to validate contents of the zip file
```

Comment:
So, this is a behavior I didn't replicate because it didn't make sense to me. The current unzip uses the zip name as the root folder to unzip to. The implementation here doesn't care about the filename and only unzips the file contents.

The tests were updated to reflect this behavior change.

Reply:
The problem was that the folder was not included in the zip. This has to do with how I was querying the files in the folder and constructing the zip. The latest commits remedy this.